src/HasTags.php

```diff
@@ -217,6 +217,7 @@ protected function syncTagIds($ids, string $type = null, $detaching = true)
         $current = $this->tags()
             ->newPivotStatement()
             ->where('taggable_id', $this->getKey())
+            ->where('taggable_type', self::class)
```

Comment:
I just added a filter for the model that the foreign relates to, does that seem sufficient of solution? Would love to get this fix released in my app  soon :)